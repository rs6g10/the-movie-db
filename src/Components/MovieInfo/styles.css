.movie-details {
    border-radius: 3px;
    background: rgba(0, 0, 0, 0.85);
    opacity: 0.85;
    padding: 0;
}

.poster-container .poster {
    width: 100%;
    border-radius: 0 0 3px 3px;
    padding: 0;
}

.info-container {
    padding: 15px;
    color: #fafafa;
}

.info-container h1 {
    text-transform: uppercase;
    font-weight: bold;
    font-size: 2.2em;
}

.additional-info {
    margin: 35px 0 20px;
    margin-top: 15px;
    width: 100%;
}

.meta-data {
    display: block;
    color: #00FC87;
    font-size: 1.3em;
    line-height: 1.1em;
}

.meta-data-small {
    /*display: block;*/
    color: #00FC87;
    font-size: 0.9em;
    line-height: 1.1em;
}

.tagline {
    display: block;
    padding-bottom: 0.25em;
    color: #00FC87;
    font-size: 1.3em;
}

.nopadding {
    padding: 0;
}

@media (max-width: 767px) {
    .info-container {
        padding: 25px;
    }
}

@media (max-width: 991px) {
    .poster-container .poster {
        border-radius: 3px 0 0 3px;
    }

    .info-container {
        padding: 25px;
    }

    .info-container h1 {
        font-size: 2.5em;
    }
}
