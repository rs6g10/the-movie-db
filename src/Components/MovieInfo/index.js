import React, { Component } from 'react';
import './styles.css';
import { convertToHollywoodMoney } from '../../utility';

let backdropImage;

class MovieInfo extends Component {
  componentDidUpdate() {
    document.body.style.backgroundImage = `url(${backdropImage})`;
  }

  componentWillUnmount() {
    document.body.style.backgroundImage = '';
  }

  render() {
    const { data } = this.props;
    let posterIMG = `https://image.tmdb.org/t/p/w500${data.poster}`;
    let totalRevenue = data.revenue;
    const director = data.director && data.director.length > 0 ?
      data.director[0].name : '-';
    const stars = data.cast ? data.cast.map(x => x.name).join(', ') : '-';
    const noData = '-';
    backdropImage = `https://image.tmdb.org/t/p/original${data.backdrop}`;

    if (totalRevenue === 'undefined' || totalRevenue === 0) {
      totalRevenue = noData;
    } else {
      totalRevenue = convertToHollywoodMoney(data.revenue);
    }
    const runningTime = data.runtime ? `${data.runtime} mins` : '';

    if (data.poster == null) {
      posterIMG = 'https://i.imgur.com/dD0Zseb.png';
    }

    return (
      <div className="col-xs-12 movie-details">
        <div className="container row">
          <div
            className="poster-container col-xs-12 col-md-4 pull-md-8 col-lg-5 pull-lg-6"
          >
            <img
              id="poster"
              className="poster"
              src={posterIMG}
              alt="movie-poster"
            />
          </div>
          <div
            className="info-container col-xs-12 col-md-8 push-md-4 col-lg-7 push-lg-5"
          >
            <h1>{data.original_title}</h1>
            <span className="tagline">{data.tagline}</span>
            <p>{data.overview}</p>
          </div>
        </div>
        <div className="container">
          <div className="row info-container additional-info">
            <div className="col-sm-12">
              Director:
              <span className="meta-data-small">{director}</span>
            </div>
            <div className="col-sm-12">
              Stars:
              <span className="meta-data-small">{stars}</span>
            </div>
          </div>
          <div className="row info-container additional-info">
            <div className="col-sm-3">
              Rating:
              <span className="meta-data">{data.vote}</span>
            </div>
            <div className="col-sm-3">
              Released Date:
              <span className="meta-data">{data.release}</span>
            </div>
            <div className="col-sm-3">
              Running Time:
              <span className="meta-data">{runningTime}</span>
            </div>
            <div className="col-sm-3">
              Box Office:
              <span className="meta-data">{totalRevenue}</span>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default MovieInfo;
