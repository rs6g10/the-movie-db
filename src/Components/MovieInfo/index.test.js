import '../../test/setupEnzyme';

import React from 'react';
import { shallow } from 'enzyme';

import MovieInfo from './';

describe('MovieInfo', () => {
  it('should render as expected with valid movieDetails', () => {
    const props = {
      data: {
        backdrop: 'blah',
        cast: [],
        director: [],
        error: null,
        poster: 'poster',
        revenue: 287382987,
      },
    };
    const element = shallow(<MovieInfo {...props} />);

    expect(element).toMatchSnapshot();
  });
});
