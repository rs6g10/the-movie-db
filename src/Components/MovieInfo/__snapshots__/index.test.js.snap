// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should render as expected with valid movieDetails 1`] = `
<div
  className="col-xs-12 movie-details"
>
  <div
    className="container row"
  >
    <div
      className="poster-container col-xs-12 col-md-4 pull-md-8 col-lg-5 pull-lg-6"
    >
      <img
        alt="movie-poster"
        className="poster"
        id="poster"
        src="https://image.tmdb.org/t/p/w500poster"
      />
    </div>
    <div
      className="info-container col-xs-12 col-md-8 push-md-4 col-lg-7 push-lg-5"
    >
      <h1 />
      <span
        className="tagline"
      />
      <p />
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row info-container additional-info"
    >
      <div
        className="col-sm-12"
      >
        Director:
        <span
          className="meta-data-small"
        >
          -
        </span>
      </div>
      <div
        className="col-sm-12"
      >
        Stars:
        <span
          className="meta-data-small"
        />
      </div>
    </div>
    <div
      className="row info-container additional-info"
    >
      <div
        className="col-sm-3"
      >
        Rating:
        <span
          className="meta-data"
        />
      </div>
      <div
        className="col-sm-3"
      >
        Released Date:
        <span
          className="meta-data"
        />
      </div>
      <div
        className="col-sm-3"
      >
        Running Time:
        <span
          className="meta-data"
        />
      </div>
      <div
        className="col-sm-3"
      >
        Box Office:
        <span
          className="meta-data"
        >
          $ 82588981217 Million
        </span>
      </div>
    </div>
  </div>
</div>
`;
